packages/compiler-sfc/src/style/pluginScoped.ts

```diff
@@ -216,17 +217,21 @@ function rewriteSelector(
           return false
         }
       }
-      // .foo * -> .foo[xxxxxxx] *
-      if (node) return
+      //                         v- store this
+      // .foo * -> .foo[xxxxxxx] *[xxxxxxx]
```

Comment:
```suggestion
      // .foo * -> .foo[xxxxxxx] [xxxxxxx]
```

Reply:
While yes, in the final output the star will be removed, in my opinion the intent of that part of the code is clearer if it's included there. We're storing the star so it can have the scoping tag added to it at the end of the function. The fact that we then delete the redundant star isn't relevant to what we're conceptually doing here.